### ISSUE TITLE
SQLite support

### ISSUE THREAD

**Issue by rgov** [google/souper#224]
I'd like to be able to use SQLite instead of Redis as the cache key-value store.

This would allow one to treat the cache as a build product that can be deleted, re-used, or deployed to another machine.

SQLite does allow concurrent access from multiple processes.

The same thing can be accomplished with Redis via DUMP after a build and FLUSHDB, RESTORE before a build, as long as there aren't simultaneous builds.

**Comment by regehr** [comment]
This probably isn't too difficult but I'm afraid I don't see the point yet of doing this.

**Comment by rgov** [comment]
I'm using a build environment which is running on someone else's servers (which I cannot install new software on), and which is sandboxed to forbid connections to network services.

The workflow I want to support is:

1. Upload sources + Souper + pre-filled cache to build machine.
2. Build.
3. Download build products + cache.

You can imagine doing this across many isolated build machines, and having a fourth "merge caches" step.

sqlite3 is also pre-installed on my platform and so I could skip building hiredis as a dependency.

**Comment by regehr** [comment]
I agree this doesn't sound like much fun, but if you're just playing with Souper, why not find a random i7 that you have full control over and work there?

I'd likely be positive about a pull request supporting SQLite but I'm probably not going to have time to do this sort of change myself in the near future.
